domain/src/main/java/org/oppia/domain/topic/TopicController.kt

```diff
@@ -545,7 +745,7 @@ class TopicController @Inject constructor(
       .setExplorationId(TEST_EXPLORATION_ID_30)
       .setName("Prototype Exploration")
       .setSummary("This is the prototype exploration to verify interaction functionality.")
-      .setChapterPlayState(ChapterPlayState.NOT_STARTED)
+      .setChapterPlayState(ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES)
```

Comment:
Shouldn't the first chapter of a story be NOT_STARTED if there are no prior chapters to complete? The missing prerequisites case was meant for when a later chapter cannot yet be started because a previous one isn't yet completed (ie, the first chapter not yet started that can be completed should be NOT_STARTED; everything before should be COMPLETED, everything after should be NOT_PLAYABLE_MISSING_PREREQUISITES).

Reply:
Actually, this progress is getting updated in `getStory` when we are using the progress data provider to combine. I have done this, so that we can set everything by default as `NOT_PLAYABLE_MISSING_PREREQUISITES ` and the combine function will take care of everything using this code:

```
/** Helper function for [combineTopicAndTopicProgress] to set first chapter as NOT_STARTED in [StorySummary]. */
  private fun setFirstChapterAsNotStarted(storySummary: StorySummary): StorySummary {
    val chapterBuilder = storySummary.getChapter(0).toBuilder()
    chapterBuilder.chapterPlayState = ChapterPlayState.NOT_STARTED
    val storyBuilder = storySummary.toBuilder()
    return storyBuilder.setChapter(0, chapterBuilder).build()
  }
```

Reply:
Ah I see, but does this mean that the eventual implementation that populates these objects from Oppia backend’s data models also has to default as missing prerequisites? To keep things simpler, maybe default to the proto default and design the progress controller to set/infer missing prerequisites. Otherwise the contract of this data model gets a bit complicated to set up correctly, and failing to do so may result in unexpected behavior.

Reply:
From backend we are getting this information:
```
"story_contents": {
          "nodes": [
            {
              "exploration_id": "umPkwp0L1M0-",
              "destination_node_ids": [
                "node_2"
              ],
              "prerequisite_skill_ids": [],
              "title": "What is a Fraction?",
              "outline_is_finalized": false,
              "outline": "",
              "id": "node_1",
              "acquired_skill_ids": []
            },
            {
              "exploration_id": "MjZzEVOG47_1",
              "destination_node_ids": [],
              "prerequisite_skill_ids": [],
              "title": "The Meaning of Equal Parts",
              "outline_is_finalized": false,
              "outline": "",
              "id": "node_2",
              "acquired_skill_ids": []
            }
          ],
          "initial_node_id": "node_1",
          "next_node_id": "node_3"
        }
```

In this, `destination_node_ids` is managing this **NOT_STARTED** and ** NOT_PLAYABLE_MISSING_PREREQUISITES** but actually in android, if we are using this to populate the initial data, then its fine to change our implementation to previous but the code that I have written just makes sure that `index 0` is **NOT_STARTED** and rest are **NOT_PLAYABLE_MISSING_PREREQUISITES**. and if the data layer is keep track of that, then its fine, we can move back to previous implementation.

Reply:
Right I understand that, but my point is actually that this now puts a constraint on future code that it needs to properly set ``NOT_PLAYABLE_MISSING_PREREQUISITES`` for all states except the first/completed ones, which I'm not sure is what upstream code is expecting to do.

What I expect is that when we download the explorations from the backend, we'll know if they've been finished or not. That's all the information we have--missing prerequisites is an inferred, Android-only state. Subsequently, the topic controller I think needs to be the thing that infers this, otherwise we have to properly compute the 'latest' state in two places (once where we populate the future topic database and second when we compute progress here).

You're right that we're hardcoding things, but we need to design for the end state and not for the current state. If you think that upstream code that downloads and sets up the topics themselves, once it exists, should also manage initializing everything properly to missing prerequisites/not started, let's chat because then we are in disagreement and I'd like to resolve that.